src/session_impl.cpp

```diff
@@ -6500,6 +6526,24 @@ namespace aux {
 		m_alerts.emplace_alert<dht_pkt_alert>(pkt, len, d, node);
 	}
 
+	bool session_impl::on_dht_request(char const* query, int query_len
+		, dht::msg const& request, entry& response)
+	{
+#ifndef TORRENT_DISABLE_EXTENSIONS
+		if (query_len > max_dht_query_length) return false;
+
+		for (m_extension_dht_queries_t::iterator i = m_extension_dht_queries.begin();
```

Comment:
please make sure this is efficient. With just a few extensions and a hundred messages per second it doing a linear scan could easily become expensive for instance. Is this called only for unrecognized messages?